helpers/compile/plugins/fill-plugin/fillers/buffer-small.ts

```diff
@@ -0,0 +1,504 @@
+    })
+  }
+
+  toString(encoding: Encoding = 'utf8', start = 0, end = this.length) {
+    start = start < 0 ? 0 : start
+    encoding = encoding.toString().toLowerCase() as Encoding
+
+    if (end <= 0) return ''
+
+    if (encoding === 'utf8' || encoding === 'utf-8') {
+      return decoder.decode(this.slice(start, end))
+    }
+    if (encoding === 'base64' || encoding === 'base64url') {
+      const string = btoa(this.reduce((s, v) => s + c2s(v), ''))
+
+      if (encoding === 'base64url') {
+        return string.replace(/\+/g, '-').replace(/\//g, '_').replace(/=+$/, '')
+      }
+
+      return string
+    }
+    if (encoding === 'binary' || encoding === 'ascii' || encoding === 'latin1' || encoding === 'latin-1') {
+      return this.slice(start, end).reduce((s, v) => s + c2s(v & (encoding === 'ascii' ? 0x7f : 0xff)), '')
+    }
+    if (encoding === 'ucs2' || encoding === 'ucs-2' || encoding === 'utf16le' || encoding === 'utf-16le') {
+      let string = ''
+      const view = new DataView(this.slice(start, end).buffer)
+      for (let i = 0; i < view.byteLength; i += 2) {
+        string += c2s(view.getUint16(i, true))
+      }
+      return string
+    }
+    if (encoding === 'hex') {
+      return this.slice(start, end).reduce((s, v) => s + v.toString(16).padStart(2, '0'), '')
+    }
+
+    bufferPolyfillDoesNotImplement(`encoding "${encoding}"`)
+  }
+
+  toLocaleString() {
+    return this.toString()
+  }
+}
+
+function stringToBuffer(value: string, encoding: string) {
+  encoding = encoding.toLowerCase() as Encoding
+
+  if (encoding === 'utf8' || encoding === 'utf-8') {
+    return new Buffer(encoder.encode(value))
+  }
+  if (encoding === 'base64' || encoding === 'base64url') {
+    return new Buffer([...atob(value)].map((v) => v.charCodeAt(0)))
+  }
+  if (encoding === 'binary' || encoding === 'ascii' || encoding === 'latin1' || encoding === 'latin-1') {
+    return new Buffer([...value].map((v) => v.charCodeAt(0)))
+  }
+  if (encoding === 'ucs2' || encoding === 'ucs-2' || encoding === 'utf16le' || encoding === 'utf-16le') {
+    const buffer = new Buffer(value.length * 2)
+    const view = new DataView(buffer.buffer)
+    for (let i = 0; i < value.length; i++) {
+      view.setUint16(i * 2, value.charCodeAt(i), true)
+    }
+    return buffer
+  }
+  if (encoding === 'hex') {
+    const bytes = new Buffer(value.length / 2)
+    for (let byteIndex = 0, i = 0; i < value.length; i += 2, byteIndex++) {
+      bytes[byteIndex] = parseInt(value.slice(i, i + 2), 16)
+    }
+
+    return bytes
+  }
+
+  bufferPolyfillDoesNotImplement(`encoding "${encoding}"`)
+}
+
+const encoder = new TextEncoder()
+const decoder = new TextDecoder()
+
+export type Encoding =
+  | 'utf8'
+  | 'utf-8'
+  | 'hex'
+  | 'base64'
+  | 'ascii'
+  | 'binary'
+  | 'base64url'
+  | 'ucs2'
+  | 'ucs-2'
+  | 'utf16le'
+  | 'utf-16le'
+  | 'latin1'
+  | 'latin-1'
+
+function initReadMethods(prototype: Buffer) {
```

Comment:
This should be called `createDataViewMethods`. It dynamically creates all the methods for the types that you saw on line 2.